### PR TITLE
Add item description page

### PR DESCRIPTION
**Pull request by Ostapenco**
@dumitru-stratulat

### CODE REVIEW

File: next-env.d.ts [hunk 1]
```diff
@@ -6,4 +6,14 @@ declare module "*.module.css" {
   export default classes;
 }
 
-type Status = "pending" | "success" | "error";
+type StatusState = {
```
Comment (AleVul) [inline next-env.d.ts:9]:
```suggestion
type StatusState = { status: "idle" } |  { status: "pending" } |  { status: "success" } |  { status: "error", message: string };
```

File: next-env.d.ts [hunk 2]
```diff
@@ -6,4 +6,14 @@ declare module "*.module.css" {
...
+  status: "idle" | "pending" | "success" | "error";
+  message?: unknown;
+};
+
+type ProductData = {
+  title: title;
+  price: number;
+  content: content;
+  imagesUrl: [unknown];
```
Comment (AleVul) [inline next-env.d.ts:18]: why not `string[]`?

File: src/pages/login/index.tsx [hunk 3]
```diff
@@ -13,87 +13,108 @@ interface Value {
...
+    "Write it down somewhere next time!",
+    "Try to remember!",
+    "Not my problem",
+    "Stop clicking me!!!",
+    "Ok, Let's try to guess together",
+    "Have you tried 11111?",
+    "Maybe 12345?",
+    "Then I have no clue...",
+    "Here we go again",
+  ];
```
Comment (AleVul) [inline src/pages/login/index.tsx:50]: what is this? :)

File: src/pages/login/index.tsx [hunk 4]
```diff
@@ -13,87 +13,108 @@ interface Value {
...
-            <a href="">register now!</a>
-          </Link>
-        </Form.Item>
-        {status === "error" && (
-          <h4 className={style.error}>Oops, something went wrong!</h4>
-        )}
-      </Form>
+  return (
+    <div className={style.loginFormContainer}>
+      {statusState.status === "pending" ? (
```
Comment (AleVul) [inline src/pages/login/index.tsx:111]: you could just render Spin with loginForm as child node and set `spinning` prop on `Spin` equal to `statusState.status === "pending"`

File: src/pages/product/[id].tsx [hunk 5]
```diff
@@ -1,16 +1,83 @@
...
+        const response = await axios.get(
+          "https://reactive.loca.lt/product/5f573c687a3d7f23509b52fd",
+          {
+            headers: {
+              Authorization: `Bearer ${localStorage.getItem("token")}`,
+            },
+          }
+        );
+        setData(response.data[0]);
+        console.log("getProduct -> response.data[0]", response.data[0]);
```
Comment (AleVul) [inline src/pages/product/[id].tsx:28]: please remove log

File: src/pages/product/[id].tsx [hunk 6]
```diff
@@ -1,16 +1,83 @@
...
+        setStatusState({ status: "success" });
+      } catch (err) {
+        setStatusState({ status: "error", message: err.response.data.message });
+        console.log(
```
Comment (AleVul) [inline src/pages/product/[id].tsx:32]: same, please remove log

File: src/pages/product/[id].tsx [hunk 7]
```diff
@@ -1,16 +1,83 @@
...
+        </span>
+        <Button type="primary" htmlType="submit" className={style.loginButton}>
+          Buy
+        </Button>
+      </div>
+    </div>
+  );
+
+  console.log(statusState.status);
+  console.log(data);
```
Comment (AleVul) [inline src/pages/product/[id].tsx:68]: same

File: src/pages/product/[id].tsx [hunk 8]
```diff
@@ -1,16 +1,83 @@
...
+import style from "./product.module.css";
 
 export default function Product() {
-  const router = useRouter();
+  const [data, setData] = useState<ProductData | null>();
+  const [statusState, setStatusState] = useState<StatusState>({
+    status: "idle",
+  });
+
+  useEffect(() => {
```
Comment (dumitru-stratulat) [inline src/pages/product/[id].tsx:15]: please implement next js

**Pull request closed without merging**